src/Crypto/Encoding/BIP39.hs

```diff
@@ -290,3 +300,140 @@ phraseToSeed mw dic passphrase =
+-- with the given 'Dictionary'.
+--
+-- This function assumes the 'Dictionary' and the 'MnemonicPhrase' are
+-- compatible (see 'checkMnemonicPhrase').
+--
+mnemonicPhraseToMnemonicSentence :: forall mw . ValidMnemonicSentence mw
+                                 => Dictionary
+                                 -> MnemonicPhrase mw
+                                 -> Either DictionaryError (MnemonicSentence mw)
+mnemonicPhraseToMnemonicSentence dic (MnemonicPhrase ln) = MnemonicSentence <$>
+    ListN.mapM (dictionaryWordToIndex dic) ln
+
+-- | convert the given generic 'MnemonicSentence' to a human readable
+-- 'MnemonicPhrase' targetting the language of the given 'Dictionary'.
+mnemonicSentenceToMnemonicPhrase :: forall mw . ValidMnemonicSentence mw
+                                 => Dictionary
+                                 -> MnemonicSentence mw
+                                 -> MnemonicPhrase mw
+mnemonicSentenceToMnemonicPhrase dic (MnemonicSentence ln) = MnemonicPhrase $
+    ListN.map (dictionaryIndexToWord dic) ln
+
+mnemonicPhraseToString :: forall mw . ValidMnemonicSentence mw
+                       => Dictionary
+                       -> MnemonicPhrase mw
+                       -> String
+mnemonicPhraseToString dic (MnemonicPhrase ln) = mconcat $
+    intersperse (dictionaryWordSeparator dic) (ListN.unListN ln)
+
+mnemonicSentenceToString :: forall mw . ValidMnemonicSentence mw
+                         => Dictionary
+                         -> MnemonicSentence mw
+                         -> String
+mnemonicSentenceToString dic = mnemonicPhraseToString dic
+                             . mnemonicSentenceToMnemonicPhrase dic
+
+-- | translate the given 'MnemonicPhrase' from one dictionary into another.
+--
+-- This function assumes the source dictionary is compatible with the given
+-- 'MnemonicPhrase' (see 'checkMnemonicPhrase')
+--
+translateTo :: forall mw . ValidMnemonicSentence mw
+            => Dictionary -- ^ source dictionary
+            -> Dictionary -- ^ destination dictionary
+            -> MnemonicPhrase mw
+            -> Either DictionaryError (MnemonicPhrase mw)
+translateTo dicSrc dicDst (MnemonicPhrase ln) = MnemonicPhrase <$>
+    ListN.mapM (return . dictionaryIndexToWord dicDst <=< dictionaryWordToIndex dicSrc) ln
+
+------------------------------------------------------------------------
+-- Helpers
+------------------------------------------------------------------------
+
+-- | convenient type level constraint to validate a given 'Nat' e is an elemnt
+-- of the list of 'Nat' l.
+type family Elem (e :: Nat) (l :: [Nat]) :: Constraint where
+    Elem e '[] = TypeError ('Text "offset: field "
+             ':<>: 'ShowType e
+             ':<>: 'Text " not elements of valids values")
+    Elem e (e ': _) = ()
+    Elem e (_ ': xs) = Elem e xs
+
+-- Backward-compatibility for before base-4.10.0.0
+fromRight :: b -> Either a b -> b
+fromRight _ (Right b) = b
+fromRight b _         = b
```

Comment:
Not sure if it worth using pragma here? 

```hs
{-# LANGUAGE CPP #-}

{- ... -}

#if MIN_VERSION_bytestring(4, 10, 0)
import Data.Either(fromRight)
#endif

{- ... -}

#if !MIN_VERSION_bytestring(4, 10, 0)
fromRight :: b -> Either a b -> b
fromRight _ (Right b) = b
fromRight b _         = b
#endif
```

@NicolasDP, any thoughts?

Reply:
If that is only needed for a base version less that 4.10, do we still need it?

Currently cardano-sl, which as I understand it is the only client of this library only builds with ghc 8.2.2 which has base 4.10.

Reply:
Well, it is a shame to remove support for a compiler version (not that old as well) just because of a `fromRight` function. Especially since `either (const X) == fromRight X`.